### ISSUE TITLE
`vagrant ssh-config` should not be blocked by other commands

### ISSUE THREAD

**Issue by WyseNynja** [mitchellh/vagrant#5202]
I have a vagrant trigger that uses scp to fetch a file from the vagrant.  I just changed it to using `vagrant ssh-config` to connect to the instance which means I don't need to use DNS.

However, because this is a trigger and run during `vagrant provision`, I get an error:

```
==> vagrant: An action 'read_state' was attempted on the machine 'vagrant',
==> vagrant: but another process is already executing an action on the machine.
==> vagrant: Vagrant locks each machine for access by only one process at a time.
==> vagrant: Please wait until the other Vagrant process finishes modifying this
==> vagrant: machine, then try again.
```

Is there any reason that `vagrant ssh-config` blocks? Is there some way to bypass this lock?

**Comment by mitchellh** [comment]
There just isn't an easy way in the core right now to make this a reality, so it is currently blocked. There is actually another issue tracking this and we'dl ike to fix it.

**Comment by kenorb** [comment]
I believe it's fixed in #3664
